### DISCUSSION TITLE
`dunai`: Type of `reactimate` is imprecise

### DISCUSSION THREAD

**Discussion by turion** [ivanperez-keera/dunai#435] [Ideas]
It has this type signature:

```haskell
reactimate :: Monad m => MSF m () () -> m () 
```

But since `reactimate` itself continues to execute forever and never actually returns a `()`, it should have one of these type signatures:

```haskell
reactimate :: Monad m => MSF m () () -> m void
reactimate :: Monad m => MSF m () () -> m Void
```

**Comment by ivanperez-keera** [comment]
Technically, you are right: the type signature could be more precise. `Data.Void` is in `base`, so depending on it should not be a problem.

**Comment by turion** [comment]
If you want, I can send a PR :)
